cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/batch/sink/DBSink.java

```diff
@@ -168,25 +170,53 @@ private void setResultSetMetadata() throws Exception {
    *
    * @throws Exception if the driver is not available
    */
-  private void ensureJDBCDriverIsAvailable() throws Exception {
+  private void ensureJDBCDriverIsAvailable(Class<? extends Driver> jdbcDriverClass) throws Exception {
     try {
       DriverManager.getDriver(dbSinkConfig.connectionString);
     } catch (SQLException e) {
       // Driver not found. We will try to register it with the DriverManager.
       LOG.debug("Plugin Type: {} and Plugin Name: {}; Driver Class: {} not found. Registering JDBC driver via shim {} ",
-                dbSinkConfig.jdbcPluginType, dbSinkConfig.jdbcPluginName, driverClass.getName(),
+                dbSinkConfig.jdbcPluginType, dbSinkConfig.jdbcPluginName, jdbcDriverClass.getName(),
                 JDBCDriverShim.class.getName());
-      driverShim = new JDBCDriverShim(driverClass.newInstance());
-      DBUtils.deregisterAllDrivers(driverClass);
+      if (driverShim == null) {
+        driverShim = new JDBCDriverShim(jdbcDriverClass.newInstance());
+      }
+      DBUtils.deregisterAllDrivers(jdbcDriverClass);
       DriverManager.registerDriver(driverShim);
     }
   }
 
+  private boolean tableExists(Class<? extends Driver> jdbcDriverClass) {
+    try {
+      ensureJDBCDriverIsAvailable(jdbcDriverClass);
+    } catch (Exception e) {
```

Comment:
Please see my comments below for ``ensureValidConnection``, Better for this method to also throw checked exceptions instead of these ``try...catch`` blocks.